src/main/java/com/onehouse/metadata_extractor/ActiveTimelineInstantBatcher.java

```diff
@@ -0,0 +1,132 @@
+package com.onehouse.metadata_extractor;
+
+import static com.onehouse.constants.MetadataExtractorConstants.HOODIE_PROPERTIES_FILE;
+
+import com.onehouse.storage.models.File;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+import lombok.Builder;
+import lombok.Getter;
+
+public class ActiveTimelineInstantBatcher {
+
+  /**
+   * Creates batches of Hudi instants, ensuring related instants are grouped together.
+   *
+   * @param instants The list of Hudi instants.
+   * @param maxBatchSize the maximum number of instants per batch.
+   * @return A list of batches, each batch being a list of instants.
+   */
+  public List<List<File>> createBatches(List<File> instants, int maxBatchSize) {
```

Comment:
@sampan-s-nayak The logic seems error prone to me, especially for any multi-writer case.
Can we rather do this -
1. Scan through the instants in sorted order.
2. Maintain inflight and requested in a set and add instant to a list.
3. For every complete instant, we know Requested is at least needed. If requested is not present in set, halt the batch. 
4. Merge the set and the list to form the current batch.